apps/src/templates/EnhancedSafeMarkdown.jsx

```diff
@@ -0,0 +1,109 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {connect} from 'react-redux';
+
+import SafeMarkdown from './SafeMarkdown';
+import {openDialog} from '@cdo/apps/redux/instructionsDialog';
+import {renderExpandableImages} from './utils/expandableImages';
+
+export class UnconnectedExpandableImagesWrapper extends React.Component {
+  static propTypes = {
+    showImageDialog: PropTypes.func.isRequired,
+    children: PropTypes.oneOfType([
+      PropTypes.arrayOf(PropTypes.node),
+      PropTypes.node
+    ]).isRequired
+  };
+
+  componentDidMount() {
+    this.postRenderHook();
+  }
+
+  componentDidUpdate(prevProps) {
+    // TODO: do we need to do any kind of cleanup here? Or otherwise do
+    // something more precise than calling the method again when we're
+    // responding to an update rather than an initial render?
+    this.postRenderHook();
+  }
+
+  postRenderHook() {
+    const thisNode = ReactDOM.findDOMNode(this);
+    renderExpandableImages(thisNode, this.props.showImageDialog);
+  }
+
+  render() {
+    return this.props.children;
+  }
+}
+
+export const ExpandableImagesWrapper = connect(
```

Comment:
Is it worth breaking this off into a different file to help with finding it in the future? I'm not super attached as this file is already quite small.